runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -320,6 +320,9 @@ public DirectPipelineResult run(Pipeline pipeline) {
 
   private BundleFactory createBundleFactory(DirectOptions pipelineOptions) {
     BundleFactory bundleFactory = ImmutableListBundleFactory.create();
+    if (pipelineOptions.isEnforceEncodability()) {
```

Comment:
since this discards the previous bundleFactory, I think I'd prefer:

```
BundleFactory bundleFactory = pipelineOptions.isEnforceEncodability()
    ? CloningBundleFactory.create()
    : ImmutableListBundleFactory.create()
```

To make that clear.

Otherwise, I worry that we'll add more code to create/configure the bundle factory, and this will throw that away.